[MSE][GStreamer] Improve logging for TrackQueue and others

ntrrgc opened the pull request:
#### 81357e3265d4833db915bf1b600c880e07f452d3
<pre>
[MSE][GStreamer] Improve logging for TrackQueue and others
<a href="https://bugs.webkit.org/show_bug.cgi?id=243660">https://bugs.webkit.org/show_bug.cgi?id=243660</a>

Reviewed by Xabier Rodriguez-Calvar.

Add more logging code to TrackQueue and to flushing code in
SourceBufferPrivateGStreamer and WebKitMediaSrc.

No changes in functionality.

* Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.cpp:
(WebCore::SourceBufferPrivateGStreamer::flush):
* Source/WebCore/platform/graphics/gstreamer/mse/TrackQueue.cpp:
(WebCore::TrackQueue::enqueueObject):
(WebCore::TrackQueue::pop):
* Source/WebCore/platform/graphics/gstreamer/mse/WebKitMediaSourceGStreamer.cpp:
(webKitMediaSrcStreamFlush):

Canonical link: <a href="https://commits.webkit.org/253207@main">https://commits.webkit.org/253207@main</a>
</pre>

philn commented on Source/WebCore/platform/graphics/gstreamer/mse/TrackQueue.cpp:
```diff
@@ -48,7 +48,17 @@ void TrackQueue::enqueueObject(GRefPtr<GstMiniObject>&& object)
     ASSERT(GST_IS_SAMPLE(object.get()) || GST_IS_EVENT(object.get()));
 
-    GST_TRACE("TrackQueue for '%s': Putting object in the queue: %" GST_PTR_FORMAT ". notEmptyCallback currently %s.", m_trackId.string().utf8().data(), object.get(), m_notEmptyCallback ? "set, will be called" : "unset");
+#ifndef GST_DISABLE_GST_DEBUG
```
I don't think a ifdef is needed here, we don't do extra allocations or costly operations here that would be avoided by disabling GST_DEBUG, afaik.

philn commented on Source/WebCore/platform/graphics/gstreamer/mse/TrackQueue.cpp:
```diff
@@ -93,6 +103,15 @@ GRefPtr<GstMiniObject> TrackQueue::pop()
 {
     ASSERT(!isEmpty());
     GRefPtr<GstMiniObject> object = m_queue.takeFirst();
+#ifndef GST_DISABLE_GST_DEBUG
```
ditto

ntrrgc committed:
[MSE][GStreamer] Improve logging for TrackQueue and others
https://bugs.webkit.org/show_bug.cgi?id=243660

Reviewed by Xabier Rodriguez-Calvar.

Add more logging code to TrackQueue and to flushing code in
SourceBufferPrivateGStreamer and WebKitMediaSrc.

No changes in functionality.

* Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.cpp:
(WebCore::SourceBufferPrivateGStreamer::flush):
* Source/WebCore/platform/graphics/gstreamer/mse/TrackQueue.cpp:
(WebCore::TrackQueue::enqueueObject):
(WebCore::TrackQueue::pop):
* Source/WebCore/platform/graphics/gstreamer/mse/WebKitMediaSourceGStreamer.cpp:
(webKitMediaSrcStreamFlush):

Canonical link: https://commits.webkit.org/253207@main
